eth/tracers/api_blockResult.go

```diff
@@ -0,0 +1,342 @@
+package tracers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"runtime"
+	"sync"
+
+	"github.com/scroll-tech/go-ethereum/common"
+	"github.com/scroll-tech/go-ethereum/common/hexutil"
+	"github.com/scroll-tech/go-ethereum/core"
+	"github.com/scroll-tech/go-ethereum/core/state"
+	"github.com/scroll-tech/go-ethereum/core/types"
+	"github.com/scroll-tech/go-ethereum/core/vm"
+	"github.com/scroll-tech/go-ethereum/log"
+	"github.com/scroll-tech/go-ethereum/rpc"
+	"github.com/scroll-tech/go-ethereum/trie/zkproof"
+)
+
+type TraceBlock interface {
+	GetBlockResultByNumberOrHash(ctx context.Context, blockNrOrHash rpc.BlockNumberOrHash, config *TraceConfig) (trace *types.BlockResult, err error)
+}
+
+type environment struct {
+	config *TraceConfig
+
+	rMu      sync.Mutex
```

Comment:
add doc to every mutex and explain what state this mutex guards.

Reply:
added.